addons/hr_holidays/static/src/components/day_month_selection/day_month_selection.js

```
@@ -0,0 +1,36 @@
+import { registry } from "@web/core/registry";
+import { SelectionField, selectionField } from "@web/views/fields/selection/selection_field";
+
+export class DayMonthSelectionField extends SelectionField {
+    static props = {
+        ...SelectionField.props,
+        selectedMonth: { type: String, optional: true },
+    };
+
+
+    get options() {
+        const selectedMonth = this.props.record.data[this.props.selectedMonth];
+        if (isNaN(selectedMonth) || selectedMonth < 1 || selectedMonth > 12){
+            alert("Please select a valid month.");
+            return;
+        }
```

Using alert is not really standard for Odoo. You could raise an error if you want to do that.

What I had meant was to just check if it exists before the condition

---

updated